Sure, will do. The changes for NRF and MIMXRT look fine, but I cannot test it with GCC16. The most recent one I can get for Debian is GCC 15.3. With that one, build fails for devices with MIMXRT1062 and MIMXRT1064 with errors in mbedtls like:
`../../lib/mbedtls/library/common.h:343:17: error: array subscript 16 is outside array bounds of 'uint64_t[2]' {aka 'long long unsigned int[2]'} [-Werror=array-bounds=]
`.

Edit: Looking at that code in detail, the compiler complaint seems wrong. It does not consider the cast from uint64 to uint8.

--- a/ports/mimxrt/machine_can.c
+++ b/ports/mimxrt/machine_can.c
@@ -491,12 +491,7 @@ static void machine_can_port_set_filter_done(machine_can_obj_t *self) {
 // Update interrupt configuration based on the new contents of 'self'
 static void machine_can_update_irqs(machine_can_obj_t *self) {
     struct machine_can_port *port = self->port;
-    uint16_t triggers = self->mp_irq_trigger;
-    uint64_t irq_flags = 0;
     uint64_t irq_tx_mask = ~((1ULL << port->flexcan_txmb_start) - 1);
-    if (triggers & MP_CAN_IRQ_RX) {
-        irq_flags |= kFLEXCAN_RxFifoFrameAvlFlag;
-    }
     // Clear all pending MB interrupt flags
     FLEXCAN_ClearMbStatusFlags(port->can_inst,
         irq_tx_mask | (kFLEXCAN_RxFifoOverflowFlag | kFLEXCAN_RxFifoWarningFlag | kFLEXCAN_RxFifoFrameAvlFlag));

--- a/ports/renesas-ra/ra/ra_adc.c
+++ b/ports/renesas-ra/ra/ra_adc.c
@@ -332,7 +332,7 @@ void ra_adc_set_resolution(uint8_t res) {
         adcer = adc_reg->ADCER;
         adcer &= (uint16_t) ~0x0006;
         adcer |= (uint16_t)adprc;
-        adc_reg->ADCER;
+        adc_reg->ADCER = adcer;
         resolution = res;
     }
     #else
@@ -347,7 +347,7 @@ void ra_adc_set_resolution(uint8_t res) {
         adcer = adc_reg->ADCER;
         adcer &= (uint16_t) ~0x0006;
         adcer |= (uint16_t)adprc;
-        adc_reg->ADCER;
+        adc_reg->ADCER = adcer;
         resolution = res;
     }
     #endif

--- a/ports/stm32/Makefile
+++ b/ports/stm32/Makefile
@@ -372,6 +372,10 @@ HAL_SRC_C += $(addprefix $(STM32LIB_HAL_BASE)/Src/stm32$(MCU_SERIES)xx_,\
 	hal_pcd_ex.c \
 	ll_usb.c \
 	)
+ifeq ($(MCU_SERIES),$(filter $(MCU_SERIES),f4 f7))
+# HAL F4-1.16.0 and F7-1.7.0 have an unused debug variable in USB_ActivateDedicatedEndpoint.
+$(BUILD)/$(STM32LIB_HAL_BASE)/Src/stm32$(MCU_SERIES)xx_ll_usb.o: CFLAGS += -Wno-error=unused-but-set-variable
+endif
 endif
 
 ifeq ($(MCU_SERIES),$(filter $(MCU_SERIES),n6))

